web/src/app/forms/FormField.js

```diff
@@ -109,10 +109,11 @@ export class FormField extends React.PureComponent {
       ...otherFieldProps
     } = this.props
 
-    const InputLabelProps = {
+    let InputLabelProps = {
       required: required && !optionalLabels,
       ..._inputProps,
     }
+    if (this.props.value) InputLabelProps = { ...InputLabelProps, shrink: true }
```

Comment:
Instead of rebuilding the object again, we can evaluate in the initial creation:
```
const InputLabelProps = {
  required: required && !optionalLabels,
  shrink: Boolean(this.props.value)
  ..._inputProps,
}
```